samples/server.c

```diff
@@ -605,14 +656,25 @@ int main(int argc, char *argv[])
                     err(EXIT_FAILURE, "vfu_irq_trigger() failed");
                 }
 
+                printf("doing dma io\n");
+
+                /*
+                 * We initiate some dummy DMA by directly accessing the client's
+                 * memory. In this case, we keep track of dirty pages ourselves,
+                 * as the client has no knowledge of what and when we have
+                 * written to its memory.
+                 */
+                do_dma_io(vfu_ctx, &server_data, 1, false);
+                
                 /*
-                 * We also initiate some dummy DMA via an explicit message,
-                 * again to show how DMA is done. This is used if the client's
-                 * RAM isn't mappable or the server implementation prefers it
-                 * this way.  Again, the client expects the server to send DMA
-                 * messages right after it has triggered the IRQs.
+                 * We also do some dummy DMA via explicit messages to show how
```

Comment:
I think it would bit a bit more representative to do messaged-based DMA writes to half of the regions and mmap-based to the other half. You can split the regions however you like, e.g. first half vs. second half, or even-based/odd-based. This way you won't have to abuse the `flags` field with `CLIENT_DIRTY_*`, right?

Reply:
I was using `CLIENT_DIRTY_*` to let the client track dirty pages, how else would the client track this?